spec/models/user_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations for User model' do
+    before(:each) do
+      @user = User.new(name: 'Tom', bio: 'Teacher from Mexico', post_counter: 0)
+    end
+
+    before { @user.save }
+
+    it 'if there is name' do
+      @user.name = nil
+      expect(@user).to_not be_valid
+    end
+
+    it 'PostCounter must be greater than or equal to zero' do
+      @user.post_counter = -1
+      expect(@user).to_not be_valid
+    end
+
+    it 'PostCounter must be greater than or equal to zero' do
+      @user.post_counter = 5
+      expect(@user).to be_valid
+    end
```

Comment:
- [ ] Also It is suggested to write unit tests for the custom methods in your models. For example in your User model, there are `recent_posts` method, please add tests for it here 😃 👍